app/controllers/admin/ex_links_controller.rb

```diff
@@ -0,0 +1,135 @@
+# encoding:UTF-8
+class Admin::ExLinksController < ApplicationController
+  load_permissions_and_authorize_resource
+  before_action :authorize
+  before_action :set_ex_link, only: [:show, :edit, :update, :destroy]
+
+  def index
+    if params[:tag]
+      @ex_links = get_exlinks_if_all_tags_present(params[:tag])
+    else
+      @ex_links = ExLink.all
+    end
+    @tags = get_tags
+  end
+
+  def show
+    @ex_link = ExLink.find(params[:id])
+  end
+
+  def new
+    @ex_link = ExLink.new
+  end
+
+  def edit
+  end
+
+  def create
+    @ex_link = ExLink.new(ex_link_params)
+    if @ex_link.save
+      redirect_to admin_ex_link_path(@ex_link), notice: alert_create(ExLink)
+    else
+      render :new, status: 422
+    end
+  end
+
+  def update
+    if @ex_link.update(ex_link_params)
+      redirect_to admin_ex_link_path(@ex_link), notice: alert_update(ExLink)
+    else
+      render :edit, status: 422
+    end
+  end
+
+  def destroy
+    @ex_link.destroy
+    redirect_to admin_ex_links_path, notice: alert_destroy(ExLink)
+  end
+
+  # QUERY_BY_TAGS
+
+  # get all ExLinks with specific tag
+  def get_exlinks_if_tag_present(tagname)
+    ExLink.includes(:tags).where(tags: { tagname: tagname })
```

Comment:
This should probably be in exlink.rb instead. A protip (tm) is to make it into a scope because then you can add more query parts to it later.